code/modules/recycling/conveyor.dm

```diff
@@ -395,6 +396,27 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
 		icon_state = "[base_icon_state]-[invert_icon ? "rev" : "fwd"]"
 	return ..()
 
+/obj/machinery/conveyor_switch/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+	. = ..()
+	if(!held_item)
```

Comment:
just early return `if(held_item)`, won't need to optional chain everything past here

Reply:
my mistake, i mean returning after this block. this should be changed to `if(isnull(held_item))` btw as its more intended for typed items

Reply:
There is a !held_item early return to the line 401 if block on line 405. Just to clarify did you mean adding a separate if(held_item) condition and using a different pattern for the later if statements, adding a return to the entire proc after all the early returns, or something else?